### PR TITLE
Feature/54232 Ps Report - manual execution

### PR DESCRIPTION
**Pull request by GuyHarwood**
manually execute the Ps Report from the tech support area.

### CODE REVIEW

File: admin/controllers/tech-support.js [hunk 1]
```diff
@@ -492,11 +493,11 @@ const controller = {
   },
 
   postCheckResultsResyncAll: async function postCheckResultsResyncAll (req, res, next) {
-    res.locals.pageTitle = 'Resync All Results'
+    res.locals.pageTitle = 'Resync Uncalculated Results'
```
Comment (jon-shipley) [inline admin/controllers/tech-support.js:496]: It's not really "uncalculated" though is it?  More like 'Un-synchronised".

Even better, as the role is tech support here, perhaps something like "Synchronise new checks from Table Storage to Azure SQL".  


If the checkbox is ticked then the historical stuff is resynchronised too.

File: admin/controllers/tech-support.js [hunk 2]
```diff
@@ -552,6 +553,41 @@ const controller = {
...
     }
+  },
+
+  getPsReportRun: async function getPsReportRun (req, res, next) {
+    try {
+      res.locals.pageTitle = 'PS Report Run'
+      req.breadcrumbs('PS Report Run')
+      res.render('tech-support/ps-report-run', {
+        breadcrumbs: req.breadcrumbs(),
+        runReport: req.body?.runReport ?? false,
```
Comment (jon-shipley) [inline admin/controllers/tech-support.js:564]: So this will be either a string (possibly empty) or a boolean. I don't like giving it more than one type.  Is there a reason for it?

File: admin/controllers/tech-support.js [hunk 3]
```diff
@@ -552,6 +553,41 @@ const controller = {
...
+        response: ''
+      })
+    } catch (error) {
+      return next(error)
+    }
+  },
+
+  postPsReportRun: async function postPsReportRun (req, res, next) {
+    const runReport = req.body.runReport === 'true'
+    const response = runReport === true ? 'PS Report Requested' : 'Report NOT Requested - checkbox tick required'
```
Comment (jon-shipley) [inline admin/controllers/tech-support.js:574]: I don't see why you are mixing types here.  `runReport` is sometimes a string and sometimes a bool.   If we need a boolean flag for the server logic, then I would prefer a separate variable that is always a bool that is derived from the incoming form post.

File: admin/services/tech-support/ps-report-exec/ps-report-exec.data.service.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      displayName: record.displayName
+    }
+  }
+
+  public static async sendPsReportExecMessage (message: IExecPsReportRequest): Promise<void> {
+    if (this.sbClient === undefined || this.sbSender === undefined) {
+      this.sbClient = new ServiceBusClient(config.ServiceBus.connectionString)
+      this.sbSender = this.sbClient.createSender('ps-report-exec')
+    }
+    console.dir(message)
```
Comment (jon-shipley) [inline admin/services/tech-support/ps-report-exec/ps-report-exec.data.service.ts:49]: debug?

File: admin/controllers/tech-support.js (hunk 1 above)
Comment (GuyHarwood) [reply admin/controllers/tech-support.js:496]: yeah doesn't need to be high level, and makes more sense to us.
I've changed the checkbox to `Include checks that were previously synchronised`

File: admin/controllers/tech-support.js (hunk 2 above)
Comment (GuyHarwood) [reply admin/controllers/tech-support.js:564]: it's a string value, as its the state of the checkbox HTML element.  I've tried the various combinations and it has to be parsed from its string value to work.  I've updated that line to reflect this

File: admin/controllers/tech-support.js (hunk 3 above)
Comment (GuyHarwood) [reply admin/controllers/tech-support.js:574]: i am creating a boolean and setting its state based upon the string value of the checkbox element.
It does read a bit wordy, so i'll trim it back and just use the form element string value.

File: admin/services/tech-support/ps-report-exec/ps-report-exec.data.service.ts (hunk 4 above)
Comment (GuyHarwood) [reply admin/services/tech-support/ps-report-exec/ps-report-exec.data.service.ts]: yep, caught this earlier, already removed

File: admin/controllers/tech-support.js [hunk 5]
```diff
@@ -492,11 +493,11 @@ const controller = {
   },
 
   postCheckResultsResyncAll: async function postCheckResultsResyncAll (req, res, next) {
-    res.locals.pageTitle = 'Resync All Results'
+    res.locals.pageTitle = 'Synchronise new checks from Table Storage to main SQL database'
     const resyncAll = req.body.resyncAll === 'true' // convert string to bool- the user must tick the checkbox
     try {
       await resultsResyncService.resyncAllChecks(resyncAll)
-      req.breadcrumbs('Check Results - Resync All')
+      req.breadcrumbs('Check Results - Resync Uncalculated')
```
Comment (jon-shipley) [inline admin/controllers/tech-support.js:500]: I believe the "All" is correct here - it will resynchronise checks that are already in the db from table storage.

**Review by jon-shipley — APPROVED** [review]
LGTM
